cmake/external/eigen.cmake

```diff
@@ -1,21 +1,36 @@
 INCLUDE(ExternalProject)
 
 SET(EIGEN_SOURCE_DIR ${THIRD_PARTY_PATH}/eigen3)
-SET(EIGEN_INCLUDE_DIR ${EIGEN_SOURCE_DIR}/src/extern_eigen3)
-INCLUDE_DIRECTORIES(${EIGEN_INCLUDE_DIR})
 
-ExternalProject_Add(
-    extern_eigen3
-    ${EXTERNAL_PROJECT_LOG_ARGS}
-    GIT_REPOSITORY  "https://github.com/RLovelett/eigen.git"
-    GIT_TAG         70661066beef694cadf6c304d0d07e0758825c10
-    PREFIX          ${EIGEN_SOURCE_DIR}
-    UPDATE_COMMAND  ""
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND     ""
-    INSTALL_COMMAND   ""
-    TEST_COMMAND      ""
-)
+INCLUDE_DIRECTORIES(${EIGEN_SOURCE_DIR}/src/extern_eigen3)
```

Comment:
```
if(WITH_AMD_GPU)
    SET(GIT_REPOSITORY "https://github.com/sabreshao/hipeigen.git")
    SET(GIT_TAG 0cba03ff9f8f9f70bbd92ac5857b031aa8fed6f9)
else()
    SET(GIT_REPOSITORY  "https://github.com/RLovelett/eigen.git")
    SET(GIT_TAG         0cba03ff9f8f9f70bbd92ac5857b031aa8fed6f9)
endif()
ExternalProject_Add(
    extern_eigen3
    ${EXTERNAL_PROJECT_LOG_ARGS}
     GIT_REPOSITORY ${GIT_REPOSITORY}
     GIT_TAG ${GIT_TAG}
     ...
)
```
may be more cleaner

Reply:
In AMD GPU build, use hipeigen instead since it includes necessary patches for HIP.